jfr-daemon/src/main/java/com/newrelic/jfr/daemon/EventConverter.java

```diff
@@ -118,4 +118,17 @@ private void convertAndBuffer(BufferedTelemetry batches, RecordedEvent event) {
           e);
     }
   }
+
+  /** To be used as a counter when an Object is needed, as Integer is immutable. */
+  private static class IncrementableInteger {
```

Comment:
What are the benefits of this custom class versus, say, using something like `AtomicInteger.getAndIncrement()`?

Reply:
I went back and forth whether to use Atomic or Incrementable.
I saw a benchmark that gave a little edge for IncrementableInteger.

But the main reason was that I wasn't sure about memory overhead of AtomicInteger.
But looking at the code again, all the extra memory is on the stack, not the heap.

So in the spirit of using less memory, I will change this to AtomicInteger, as it will save some space in the metaspace.